fix(controlcenter): TestTasksHandler_ConcurrentTasksRun flaky on TempDir cleanup

## Symptom

Surfaced intermittently during the #343 baseline work. \`make
regression\` can fail with:

\`\`\`
--- FAIL: TestTasksHandler_ConcurrentTasksRun (0.10s)
    testing.go:1464: TempDir RemoveAll cleanup:
      unlinkat .../agents/1776619091915394000: directory not empty
FAIL    github.com/alamparelli/alf/internal/controlcenter
\`\`\`

Second run passes. The failure is a cleanup race, not an assertion
failure — Go's t.TempDir auto-cleanup races with the orchestrator
goroutine that is still writing the task directory when the test
returns.

## Root cause

\`TestTasksHandler_ConcurrentTasksRun\` launches multiple
orchestrator tasks concurrently, each writing into
\`{tempDir}/agents/{taskID}/\`. The test returns once
\`POST /tasks/run\` completes for every request, but the
orchestrator's post-run write (meta.json, logs) keeps going in the
background — by the time t.Cleanup fires, a subset of task dirs
still have files being flushed, so \`RemoveAll\` hits ENOTEMPTY.

Not a data-race — the orchestrator owns those files exclusively —
but a lifecycle-ordering issue: the test completes before the
orchestrator's goroutines drain.

## Fix options

1. Drain the orchestrator goroutines before test return — e.g.
   expose a \`WaitIdle()\` on the orchestrator, call it at the end
   of the test.
2. Make the orchestrator writes synchronous when invoked from
   ConcurrentTasksRun-style tests (add a test-only flag).
3. Replace t.TempDir with a manually-managed dir and a cleanup
   loop that retries RemoveAll with backoff.

Option 1 is the right structural fix — also useful outside tests
for clean shutdown. Option 3 is the short-term patch.

## Acceptance

- [ ] \`make regression\` green on 10 consecutive runs.
- [ ] TestTasksHandler_ConcurrentTasksRun no longer leaves orphan
  files in its TempDir.

## Impact

Baseline #343 can go red on CI intermittently. Blocks nothing,
but noisy. Revisit once the runtime step (step 5) of the
0.7.9 milestone starts — orchestrator tests will move and the
race should disappear as part of the move.

Second flaky test in the same category observed during #343 work:

\`\`\`
--- FAIL: TestCancelAllStopsTasks (2.00s)
FAIL    github.com/alamparelli/alf/internal/agents    3.118s
\`\`\`

Same pattern: passes when run in isolation, fails intermittently in the full suite. Likely the same root cause — orchestrator goroutines still running past the test return boundary. Will be included in the fix scope.

--- a/internal/agents/orchestrator.go
+++ b/internal/agents/orchestrator.go
@@ -51,8 +51,9 @@ type Orchestrator struct {
 	toolRegistry    *tooling.Registry   // optional: tool schemas for API agentic loop
 	toolExecutor    *tooling.Executor   // optional: tool subprocess runner
 
-	mu       sync.Mutex
-	running  map[string]*RunningTask
+	mu        sync.Mutex
+	running   map[string]*RunningTask
+	idCounter uint64 // monotonically-increasing suffix for taskID, guarded by mu
 }
 
 // NewOrchestrator creates a new orchestrator.
@@ -81,6 +82,27 @@ func (o *Orchestrator) SetResolveProvider(fn ResolveProviderFunc) {
 func (o *Orchestrator) SetTooling(registry *tooling.Registry, executor *tooling.Executor) {
 	o.toolRegistry = registry
 	o.toolExecutor = executor
+}
+
+// WaitIdle blocks until there are no running tasks, or ctx is cancelled.
+// Used by tests and for clean shutdown to drain in-flight orchestrator
+// goroutines before tearing down their data directory.
+func (o *Orchestrator) WaitIdle(ctx context.Context) error {
+	ticker := time.NewTicker(10 * time.Millisecond)
+	defer ticker.Stop()
+	for {
+		o.mu.Lock()
+		n := len(o.running)
+		o.mu.Unlock()
+		if n == 0 {
+			return nil
+		}
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+		}
+	}
 }
 
 // Running returns a snapshot of all currently running tasks.
@@ -256,7 +278,13 @@ func (o *Orchestrator) Run(ctx context.Context, userMessage string, systemPrompt
 		teams = allTeams
 	}
 
-	taskID := fmt.Sprintf("%d", time.Now().UnixNano())
+	// Unique taskID: nanosecond timestamp + monotonic counter. UnixNano alone
+	// can collide between concurrent Run() calls scheduled in the same
+	// nanosecond, which would clobber the running map entry. See #347.
+	o.mu.Lock()
+	o.idCounter++
+	taskID := fmt.Sprintf("%d-%d", time.Now().UnixNano(), o.idCounter)
+	o.mu.Unlock()
 	agentsParent := filepath.Join(o.dataDir, "agents")
 	taskDir := filepath.Join(agentsParent, taskID)
 	os.MkdirAll(taskDir, 0o775)

--- a/internal/controlcenter/handler_tasks_test.go
+++ b/internal/controlcenter/handler_tasks_test.go
@@ -43,6 +43,22 @@ func (p *slowProvider) Invoke(ctx context.Context, _ string, _ provider.Params, 
 
 func (p *slowProvider) release() { close(p.gate) }
 
+// waitForRunning blocks until the orchestrator reports at least n running
+// tasks, or the timeout elapses. The handler's fire-and-forget goroutine may
+// not have registered with the orchestrator yet when the HTTP response
+// returns, so callers that need to drain must synchronise on this first.
+func waitForRunning(t *testing.T, orch *agents.Orchestrator, n int, timeout time.Duration) {
+	t.Helper()
+	deadline := time.Now().Add(timeout)
+	for time.Now().Before(deadline) {
+		if len(orch.Running()) >= n {
+			return
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+	t.Fatalf("timed out waiting for %d running tasks (got %d)", n, len(orch.Running()))
+}
+
 // staticTierStore is a minimal TierStore that returns a fixed config.
 type staticTierStore struct{ cfg *TiersConfig }
 
@@ -155,14 +171,19 @@ func TestTasksHandler_LaunchIsNonBlocking(t *testing.T) {
 		t.Fatal("launch blocked waiting for orchestrator to complete")
 	}
 
-	// Release the provider so the background goroutine can finish cleanly.
+	// Release the provider and drain the orchestrator so TempDir cleanup
+	// doesn't race with background file writes. See #347.
+	waitForRunning(t, orch, 1, 2*time.Second)
 	sp.release()
-	time.Sleep(50 * time.Millisecond)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := orch.WaitIdle(ctx); err != nil {
+		t.Fatalf("orchestrator did not drain: %v", err)
+	}
 }
 
 func TestTasksHandler_ConcurrentTasksRun(t *testing.T) {
 	sp := newSlowProvider()
-	defer sp.release()
 	orch := newTestOrchestrator(t, sp)
 	h := &TasksHandler{Orchestrator: orch, DataDir: t.TempDir(), ContextDir: t.TempDir(), TierStore: newTestTierStore()}
 
@@ -183,12 +204,20 @@ func TestTasksHandler_ConcurrentTasksRun(t *testing.T) {
 	}
 	wg.Wait()
 
-	// All 3 should be tracked as running.
-	// Wait briefly for goroutines to register with orchestrator.
-	time.Sleep(100 * time.Millisecond)
+	// Wait for all 3 goroutines to register with the orchestrator, then assert.
+	waitForRunning(t, orch, 3, 2*time.Second)
 	running := orch.Running()
 	if len(running) < 2 {
 		t.Errorf("expected at least 2 concurrent running tasks, got %d", len(running))
+	}
+
+	// Drain orchestrator goroutines before t.Cleanup removes TempDir, otherwise
+	// post-run meta/log writes race with RemoveAll. See #347.
+	sp.release()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := orch.WaitIdle(ctx); err != nil {
+		t.Fatalf("orchestrator did not drain: %v", err)
 	}
 }
 

